### ISSUE TITLE
pressKey() does not work properly on Firefox

### ISSUE THREAD

**Issue by PierreRott** [DevExpress/testcafe#4115]
### What is your Test Scenario?
Use `pressKey('backspace')` on Firefox.
 
### What is the Current behavior?
`pressKey('backspace')` does not work on Firefox, i.e. it doesn't remove the character that should be removed after pressing backspace.
 
### What is the Expected behavior?
`pressKey()` should accept backspace and other keys as it is on Chrome.
 
### What is your web application and your TestCafe test code?

 
<details>
<summary>Your website URL (or attach your complete example):</summary>

```html
<input id="nativeInput" type="text" (keypress)="onKeyPress($event)"/>
```

```typescript
onKeyPress(event:KeyboardEvent) {
        const eventValue = String.fromCharCode(event.keyCode);
        console.log(eventValue);
        const isValidByPattern = /^[0-1]*$/.test(eventValue);
        console.log(isValidByPattern);
        return isValidByPattern;
 }
```
</details>

<details>
<summary>Your complete test code (or attach your test files):</summary>

 
```js
test('Should test pressKey', async () => {
    await t
        .typeText(Selector('#nativeInput'), '1')
        .pressKey('backspace')
        .expect(Selector('#nativeInput').value).eql('');
});
```
</details>

<details>
<summary>Console logs:</summary>

```
// after typing '1' to the input
1 
true
// after pressing backspace
� 
false
```
</details>

### Steps to Reproduce:

1. Create an input which handles `keypress`
2. Try to use TestCafe's `pressKey('backspace')` with it
 
### Your Environment details:
 
* testcafe version:                   1.2.1
* node.js version:                    v10.16.0
* command-line arguments:  testcafe firefox test.ts
* browser name and version: Firefox 68.0.1
* platform and version:          Windows 10 Enterprise
* other:

**Comment by aleks-pro** [comment]
Hello @PierreRott ,
I couldn't reproduce the issue. I tried this with TestCafe 1.2.1, NodeJS 10.16.0, and Firefox 68.0.1. Please check the following test example: 
```JS
import { Selector } from 'testcafe';

fixture`Fixture`
    .page`https://jsfiddle.net/qmLvroy4/`;

test('Should test pressKey', async t => {
    await t
        .switchToIframe(Selector('[name="result"]'))

        .typeText(Selector('#nativeInput'), '1')
        .pressKey('backspace')
        .expect(Selector('#nativeInput').value).eql('');
});
```
Can you please modify it to show the issue in action?

**Comment by PierreRott** [comment]
JSFiddle example has been modified: 

```
const inputEl = document.getElementById('nativeInput');
inputEl.addEventListener('keypress', onKeyPress);
function onKeyPress(event) {
  const eventValue = String.fromCharCode(event.keyCode);
  console.log(eventValue);
  const isValidByPattern = /^[0-1]*$/.test(eventValue);
  console.log(isValidByPattern);

 if (!isValidByPattern) {
    event.preventDefault();
  }

  return isValidByPattern;
}
```

**Comment by AlexKamaev** [comment]
@PierreRott 
Thank you for your example, I was able to reproduce the issue.

```HTML
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title></title>
</head>
<body>
<input id="nativeInput" type="text" />

<script>
    const inputEl = document.getElementById('nativeInput');

    inputEl.addEventListener('keypress', onKeyPress);

    function onKeyPress(event) {
        const eventValue = String.fromCharCode(event.keyCode);
        console.log(eventValue);
        const isValidByPattern = /^[0-1]*$/.test(eventValue);
        console.log(isValidByPattern);

        if (!isValidByPattern) {
            event.preventDefault();
        }

        return isValidByPattern;
    }
</script>

</body>
</html>
```

```JS
import { Selector } from 'testcafe';

fixture`Fixture`
    .page`../pages/index.html`;

test('test', async t => {
    await t
        .typeText(Selector('#nativeInput'), '111')
        .pressKey('backspace')
        .pressKey('backspace')
        .pressKey('backspace')
        .expect(Selector('#nativeInput').value).eql('');
});

```

**Comment by moh0040** [comment]
Hi, This issue can be reproduce with **.pressKey('tab')** and **.pressKey('shift+left')** as well in Firefox

```html
<!DOCTYPE html>
<html>
<body>

<input id="nativeInput" type="text"/>
<input id="nativeInput2" type="text"/>

<script>
    const inputEl = document.getElementById('nativeInput');
    inputEl.addEventListener('keypress', onKeyPress);

    function onKeyPress(event) {
        const eventValue = String.fromCharCode(event.keyCode);
        const isValidByPattern = /^[0-1]*$/.test(eventValue);
        if (!isValidByPattern) {
            event.preventDefault();
        }
        return isValidByPattern;
    }
</script>

</body>
</html>

```

```typescript
import {Selector, t} from 'testcafe';

fixture`pressKey`
    .page(`index.html`);
test.only('Should test pressKey 01', async () => {
    await t
        .wait(20000)
        .click(Selector('#nativeInput'))
        .pressKey('tab')
        .expect(Selector('#nativeInput2').focused).ok();
});

test('Should test pressKey 02', async () => {
    await t
        .typeText(Selector('#nativeInput'), '1')
        .pressKey('shift+left')
        .pressKey('backspace')
        .expect(Selector('#nativeInput').value).eql('');
});
```

**Comment by AlexKamaev** [comment]
@moh0040 Thank you for your input. I was able to reproduce the incorrect behavior in Firefox.

**Comment by PierreRott** [comment]
Can you reopen the issue, please? It is still active. 
Also it is possible that it is somehow related to issues 7342, 6452, and 6969, where `pressKey` does not work as expected within Firefox browser.

**Comment by daimoonis** [comment]
Hello, is there any update regarding this issue? Seems to me the Firefox pressKey issue is still relevant. I was able to reproduce both cases (https://github.com/DevExpress/testcafe/issues/4115#issuecomment-520394678, https://github.com/DevExpress/testcafe/issues/4115#issuecomment-597091284) on testcafe 2.0.0 and FF 107.

**Comment by hendriku** [comment]
The issue was resolved for us through an update to testcafe 2.3.0
